Fix to single note widget

dan0xii opened the pull request:
Opening a note from the widget would have no effect if the user already had an editor
open for a different note (mentioned in #263). This change should check to see if a note is already being
edited, save and synchronise it and then present the new note for editing.

Unsure as to whether db.updateNoteAndSync is the best way to immediately save the note before overwriting it with the new one.

korelstar commented on app/src/main/java/it/niedermann/owncloud/notes/android/activity/EditNoteActivity.java:
```diff
@@ -75,6 +75,27 @@ protected void onCreate(final Bundle savedInstanceState) {
+        // Does the note retrieved from the intent match the note currently
+        // being displayed by the app
+        if ((note.getId()) != fragment.getNote().getId()) {
+            db.updateNoteAndSync(fragment.getNote(), fragment.getNote().getContent(), null);
```
I don't think that you need this, since your `NoteEditFragment.getNote()` saves the note automatically, when called (I don't like this, but that's how it's currently works).

korelstar commented on app/src/main/java/it/niedermann/owncloud/notes/android/activity/EditNoteActivity.java:
```diff
@@ -75,6 +75,27 @@ protected void onCreate(final Bundle savedInstanceState) {
+            originalNote = note;
+            createEditFragment(note);
+        }
+        super.onResume();
```
Is there a need, that the superclass method is called at the end? The [documentation](https://developer.android.com/guide/components/activities/activity-lifecycle.html) says: "Always call the superclass first"

dan0xii commented:
> I don't think that you need this, since your NoteEditFragment.getNote() saves the note automatically, when called (I don't like this, but that's how it's currently works).

I didn't spot that, thanks. 

> Is there a need, that the superclass method is called at the end? The documentation says: "Always call the superclass first"

That's a mistake, I wasn't aware of that. I've fixed both now and updated the branch.

korelstar commented:
Thanks!
